drivers/gpio/egpio_nrfx.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2024, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_nrfe_gpio
+
+#include <nrfx_gpiote.h>
+#include <string.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/dt-bindings/gpio/nordic-nrf-gpio.h>
+#include <zephyr/irq.h>
+
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#include <zephyr/ipc/ipc_service.h>
+
+#include <nrfe_gpio.h>
+
+K_SEM_DEFINE(bound_sem, 0, 1);
+
+static void ep_bound(void *priv)
+{
+	k_sem_give(&bound_sem);
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+}
+
+static struct ipc_ept_cfg ep_cfg = {
+	.cb = {
+		.bound    = ep_bound,
+		.received = ep_recv,
+	},
+};
+
+struct ipc_ept ep;
+
+struct gpio_nrfx_data {
```

Comment:
```suggestion
struct gpio_nrfe_data {
```
Same in other instances throughout the file.